src/OpenLoco/Windows/ScenarioSelect.cpp

```diff
@@ -0,0 +1,507 @@
+            if (scenariomgr::hasScenariosForCategory(i))
+            {
+                widget.type = widget_type::wt_8;
+                widget.left = xPos;
+                xPos += 90;
+                widget.right = xPos++;
+            }
+            else
+                widget.type = widget_type::none;
+        }
+    }
+
+    // 0x00443946
+    static void initList(window* self)
+    {
+        if (self->info == 0xFFFFFFFF)
+            return;
+
+        using namespace scenariomgr;
+        auto scenarioInfo = reinterpret_cast<ScenarioIndexEntry*>(self->info);
+        if (hasScenarioInCategory(self->current_tab, scenarioInfo))
+            return;
+
+        // Reset currently selected scenario if it is not in the current category.
+        self->info = 0xFFFFFFFF;
+        self->invalidate();
+    }
+
+    // 0x00443868
+    window* open()
+    {
+        window* self = WindowManager::bringToFront(WindowType::scenarioSelect);
+        if (self != nullptr)
+            return self;
+
+        initEvents();
+
+        self = WindowManager::createWindow(
+            WindowType::scenarioSelect,
+            Gfx::point_t({ static_cast<int16_t>(width() / 2 - windowSize.width / 2),
+                           std::max<int16_t>(height() / 2 - windowSize.height / 2, 28) }),
+            windowSize,
+            window_flags::transparent,
+            &_events);
+
+        self->widgets = _widgets;
+        self->enabled_widgets = (1 << widx::close) | (1 << widx::tab0) | (1 << widx::tab1) | (1 << widx::tab2) | (1 << widx::tab3) | (1 << widx::tab4);
+        self->initScrollWidgets();
+
+        self->colours[0] = PaletteIndex::transparent;
+        self->colours[1] = PaletteIndex::index_0B;
+
+        self->var_846 = 0xFFFF;
+        self->var_85A = static_cast<int32_t>(0xFFFFFFFF);
+
+        initTabs(self);
+
+        // Select the last tab used, or the first available one.
+        uint8_t selectedTab = config::get().scenario_selected_tab;
+        if (self->widgets[widx::tab0 + selectedTab].type == widget_type::none)
+        {
+            selectedTab = 0;
+            for (int i = 0; i < 5; i++)
+            {
+                if (self->widgets[widx::tab0 + i].type == widget_type::none)
+                {
+                    selectedTab = i;
+                    break;
+                }
+            }
+        }
+
+        self->current_tab = selectedTab;
+
+        initList(self);
+
+        return self;
+    }
+
+    // 0x00443995
+    static void prepareDraw(window* self)
+    {
+        self->activated_widgets &= ~((1 << widx::tab0) | (1 << widx::tab1) | (1 << widx::tab2) | (1 << widx::tab3) | (1 << widx::tab4));
+        self->activated_widgets |= (1ULL << (self->current_tab + static_cast<uint8_t>(widx::tab0)));
+    }
+
+    // 0x004439AF
+    static void draw(window* self, Gfx::drawpixelinfo_t* dpi)
+    {
+        Gfx::drawRectInset(dpi, self->x, self->y + 20, self->width - 1, 1, self->colours[0], 0);
```

Comment:
`drawRectInset` already takes care of the `width - 1` stuff. Also, 41 instead of 1.
```suggestion
        Gfx::drawRectInset(dpi, self->x, self->y + 20, self->width , 41, self->colours[0], 0);
```